MapboxDirections/MBRouteOptions.swift

```diff
@@ -497,20 +161,14 @@ open class RouteOptions: NSObject, NSSecureCoding, NSCopying{
         return (waypoints, routes)
     }
     
+    
     // MARK: NSCopying
-    open func copy(with zone: NSZone? = nil) -> Any {
-        let copy = RouteOptions(waypoints: waypoints, profileIdentifier: profileIdentifier)
+    override open func copy(with zone: NSZone? = nil) -> Any {
+//        let copy = RouteOptions(waypoints: waypoints, profileIdentifier: profileIdentifier)
+        let copy = super.copy() as! RouteOptions
```

Comment:
Calling super here throws the process into an infinite loop. 

![image](https://user-images.githubusercontent.com/1058624/36749177-d4886d0e-1bae-11e8-950b-ee2ec4b9f8fc.png)

@JThramer any ideas?

Reply:
What’s the stack look like further down, where the recursion begins? What causes `DirectionsOptions.copy(with:)` to reenter `RouteOptions.copy(with:)`?

Reply:
Unsure. Here is where super.copy is called: https://github.com/mapbox/MapboxDirections.swift/pull/236/files#diff-c1adc517aeca20c01aad5859dff47b21R213

Reply:
When you hit this infinite recursion, scroll down to where it stops saying `RouteOptions.copy(with:)`. What are the first few rows that are different?